Services/source/housekeeping/housekeeping_to_beacon.c

```diff
@@ -0,0 +1,162 @@
+  if (all_hk_data->EPS_startup_hk.last_reset_reason_reg = 0x24000000) {EPS_last_reset_reason = NRST;}
+  if (all_hk_data->EPS_startup_hk.last_reset_reason_reg = 0x14000000) {EPS_last_reset_reason = Software;}
+
+  //TEST: eps_last_reset_reason added, define the function based on EPS ICD manual
+  memcpy(&(beacon_packet.eps_last_reset_reason), &EPS_last_reset_reason, sizeof(uint8_t));
+  
+  /*-------Watchdog-------*/
+  //TODO: More work to be done on watchdogs
+  memcpy(&(beacon_packet.gs_wdt), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint8_t));
+  memcpy(&(beacon_packet.obc_wdt), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint16_t));
+  memcpy(&(beacon_packet.gs_wdt_expr), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint8_t));
+  memcpy(&(beacon_packet.obc_wdt_expr), &(all_hk_data->EPS_hk.outputConverterState), sizeof(uint16_t));
+  
+  /*-------Temperatures-------*/
+  //Onboard battery pack temp location defined in ICD
+  memcpy(&(beacon_packet.temps[0]), &(all_hk_data->EPS_hk.temp[8]), sizeof(int8_t));
+
+  //TODO: Athena MCU temperature should not be stored as long data type since data size can change depending on the computer
+  //      should stick with int64_t format. Change the temp driver to fix this. 
+  long athena_MCU_temp_long[2]; 
+  int8_t athena_MCU_temp[2];
+  //Min/max operating temperature on the temp sensor is -55/127 Celsius,
+  //      hence int8_t is sufficient, convert long into int8_t 
+  for (i = 0; i < 2; i++) {
+    athena_MCU_temp_long[i] = all_hk_data->Athena_hk.temparray[i];
+    if (athena_MCU_temp_long[i] > 127) {athena_MCU_temp_long[i] = 127;}
+    if (athena_MCU_temp_long[i] < -128) {athena_MCU_temp_long[i] = -128;}
+    if ((athena_MCU_temp_long[i] >> 63) == 1) { //negative
+      athena_MCU_temp[i] = ((athena_MCU_temp_long[i] & 127) | 128);
+    }
+    else {  //positive
+      athena_MCU_temp[i] = (athena_MCU_temp_long[i] & 127);
+    }
+  }
+  //use max temp out of the 2 Athena temps
+  int8_t max_MCU_temp = max(athena_MCU_temp[0], athena_MCU_temp[1]);
+  memcpy(&(beacon_packet.temps[1]), &max_MCU_temp, sizeof(int8_t));
+
+  memcpy(&(beacon_packet.temps[2]), &(all_hk_data->UHF_hk.temperature), sizeof(int8_t));
+
+  //TODO: write a function/preprocesor to determine 2U or 3U, and then fetch payload temp with CAN protocol
+  memcpy(&(beacon_packet.temps[3]), &(all_hk_data->payload_hk.payload_temp), sizeof(int8_t));
+
+  memcpy(&(beacon_packet.temps[4]), &(all_hk_data->hyperion_hk.Nadir_Temp1), sizeof(int8_t));
+  memcpy(&(beacon_packet.temps[5]), &(all_hk_data->hyperion_hk.Zenith_Temp1), sizeof(int8_t));
+  memcpy(&(beacon_packet.temps[6]), &(all_hk_data->hyperion_hk.Port_Temp1), sizeof(int8_t));
+  memcpy(&(beacon_packet.temps[7]), &(all_hk_data->hyperion_hk.Star_Temp1), sizeof(int8_t));
+
+  /*-------ADCS-------*/
+  //Of the x, y, and z angular rates, use the maximum angular rate
+  //TODO: convert angular rate from float to uint_8

```

Comment:
Could you do this

Reply:
I can't lol, looking through the ADCS manual it is unclear what the max angular rate will be. Hence I'm not sure how to change it. The ADCS team will investigate. Another clickup task for later

Reply:
Make it an issue

Reply:
we have: TODOs in the code
corresponding to: clickup tasks

do you want an issue for triple redundancy or are we okay with the above

Reply:
We'll discuss clickup vs issues at the work session. I'm not sure what the best way forward is